mesop/server/server.py

```diff
@@ -513,3 +525,16 @@ def sse_request(
         if decoded_line.startswith(SSE_DATA_PREFIX):
           event_data = json.loads(decoded_line[len(SSE_DATA_PREFIX) :])
           yield event_data
+
+
+def get_static_assets_folder() -> str | None:
```

Comment:
I think we should consider making this configurable by code instead of env var, because the static assets folder is something that almost by definition would involve code changes anyway (e.g. moving where your JS / images are).

Right now we lack any kind of top-level Mesop app configuration but I think it may be reasonable to add one, both for this use case and related to #957 we can configure app-wide default theme mode. The API could look something like this:


```py
me.configure_app(
  me.AppSettings(
    static_dir="assets", 
    static_url_path="/assets", 
    default_theme_mode="system"
))
```

Reply:
As prior art, here's how Django configures a similar feature: https://docs.djangoproject.com/en/5.1/ref/settings/#staticfiles-dirs - they also use settings and allows configuring both dir and url path (they are also much more flexible/powerful in general)

Reply:
OK, env var seems OK since we use it for everything else already.

Re: web-component dir structure, I think it depends on how people compile their web component JS code. For example, if each are standalone binaries, then your structure might make sense, but if they're compiling all their web components into 1 JS binary (e.g. to share a common UI library), then it'll be in 1 folder naturally. I think having 1 static dir to start with is good.

Reply:
Hmm, that's an interesting about point about compiling all their web components into a single JS binary. Is that possible to do now?

But couldn't they then do something like:

```
- web_component1
   - web_component1.py
   - file.js
   - file.js
   - file.scss
- web_component2
  - web_component2.py
  - file.ts
  - file.ts
  - file.css
- combined_web_component
  - web_component1.py
  - web_component2.py
  - static
     - bundle.js
     - bundle.css
```

So in this case combined_web_component, is created through some kind of build process where in the web component python files just get moved.

Granted, I'm hand-waving a bunch of stuff since moving the python files would entail changing import paths. And also is it possible to point to the same bundle.js in the web component path folder?

Reply:
I've invited you to my personal Mesop app repo - you can see how I basically compile all the web components into 1 JS binary. Besides being more convenient to setup compilation-wise, it provides a significant benefit in terms of reducing code size because then you can reuse the same libraries (e.g. code mirror) across web components. Note: when you import the same JS module multiple times, the browser automatically uses the cached import after the first time.

This is actually a pretty big issue with our current web component story, which is that if you have N web components which want to use the same library (e.g. codemirror, marked, etc), they will end up compiling it themselves _unless_ they somehow dedupe the library (e.g. CDN, or some other way of dynamically linking the library instead of compiling it in).  Other frameworks like Angular have solved this in a pretty involved way: https://angular.dev/tools/libraries/angular-package-format (not an exact parallel but vaguely resembles what we would need to do to solve this issue)

Reply:
Thanks for the context. That's good to know.